pyadjoint/checkpointing.py

```diff
@@ -164,11 +175,22 @@ def _(self, cp_action, timestep):
             ):
                 for package in self.tape._package_data.values():
                     package.continue_checkpointing()
+            if timestep == 1:
```

Comment:
I would want a big comment here explaining what is going on, perhaps even with an example. What are `_global_deps`? Why does removing `_checkpoint` do the right thing?